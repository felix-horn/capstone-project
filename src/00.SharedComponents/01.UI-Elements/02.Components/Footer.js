import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import UndoIcon from '@material-ui/icons/Undo'

Footer.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  visibilityButtonUndo: PropTypes.string,
}

export default function Footer({ onClick, className, visibilityButtonUndo }) {
  console.log({ visibilityButtonUndo })
  return (
    <FooterLayout className={className}>
      <DynamicUndoIcon
        onClick={onClick}
        visibilityButtonUndo={visibilityButtonUndo}
      />
    </FooterLayout>
  )
}

const FooterLayout = styled.div`
  height: 60px;
  max-width: 450px;
  background-color: var(--white);
  display: grid;
  place-items: center;
  box-shadow: var(--light-box-shadow-up);
`

const DynamicUndoIcon = styled(UndoIcon)`
  color: ${(props) => {
    // eslint-disable-next-line default-case
    switch (props.visibilityButtonUndo) {
      case 'hidden':
        return 'var(--white)'
      case 'grayedOut':
        return 'var(--light-gray)'
      case 'shown':
        return 'var(--dark-gray)'
    }
  }};
`
