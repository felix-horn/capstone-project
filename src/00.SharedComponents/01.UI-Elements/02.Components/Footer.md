Undo Button grayed out

```js padded
<Footer onClick={() => {}} visibilityButtonUndo="grayedOut" />
```

Undo Button grayed out

```js padded
<Footer onClick={() => {}} visibilityButtonUndo="shown" />
```
