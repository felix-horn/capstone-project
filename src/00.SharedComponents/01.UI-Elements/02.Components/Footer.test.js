import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import Footer from './Footer'

describe('Footer', () => {
  it('renders correctly', () => {
    const onClickMock = jest.fn()
    const { container } = render(
      <Footer onClick={onClickMock} visibilityButtonUndo="shown" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('calls function onClick', () => {
    const onClickMock = jest.fn()
    const { getByTestId } = render(
      <Footer onClick={onClickMock} visibilityButtonUndo="shown" />
    )
    const undoButton = getByTestId('undo-button')
    user.click(undoButton)
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })
})
