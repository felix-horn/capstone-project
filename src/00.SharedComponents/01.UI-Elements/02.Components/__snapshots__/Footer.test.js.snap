// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
.c0 {
  height: 60px;
  max-width: 450px;
  background-color: var(--white);
  display: grid;
  place-items: center;
  box-shadow: var(--light-box-shadow-up);
}

.c1 {
  color: var(--dark-gray);
}

<div
  class="c0"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root c1"
    data-testid="undo-button"
    focusable="false"
    viewBox="0 0 24 24"
    visibilityButtonUndo="shown"
  >
    <path
      d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z"
    />
  </svg>
</div>
`;
